### ISSUE TITLE
Metadata links should have link style with dotted underline

### ISSUE THREAD

**Issue by dbranchini** [sul-dlss/earthworks#1382]
For accessibility, links in a body of text cannot be indicated by color alone. For example, an underline would suffice as an additional indicator. Here's an older, but still relevant article on this topic: https://webaim.org/blog/wcag-2-0-and-link-colors/.

**Currently:**
<img src="https://github.com/user-attachments/assets/2b5a44d3-afdc-4a8a-b5e6-6fce3cdc203f" width="400">

**As designed:**
https://www.figma.com/design/FE5WZAvHf2e97GJeOP3mH6/Stanford-Styles?node-id=243-72&t=dDrF8JcHp1rG0B7M-4

Default state - Stanford Digital Blue with a dotted underline
Hover state - Stanford Digital Blue Dark with a solid underline
